Camperbot room refactor

joepurdy opened the pull request:
I condensed the camperbot rooms to a single array at @abhisekp's suggestion and removed the city rooms. I also removed additional rooms that were throwing 404s. I removed the prefixChannelName function completely since everything is hardcoded in a single array now.

Also I added FreeCodeCamp/HelpBonfires to close https://github.com/dcsan/gitterbot/issues/185

I found that the joinBonfireRooms function call in nap/lib/bot/GBot.js was causing issues with never-ending unable to join errors due to the bonfire rooms already being added in the map() function on ln 267 of nap/data/RoomData.js, so I commented it out. Not sure if this is the best approach, but it's what I've got right now to prevent the errors.

I tried to stick to the [Official List](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Official-Free-Code-Camp-Chat-Rooms) of FCC chat rooms when deciding what to keep, that being said I wasn't sure about the off topic rooms from ln 151 to ln 157 as well as some other random rooms. Theses are all the rooms in the array not included on the official list anymore:

FreeCodeCamp/40PlusDevs
FreeCodeCamp/Beta
FreeCodeCamp/BookClub
FreeCodeCamp/Design
FreeCodeCamp/Issues
FreeCodeCamp/LetsPair
FreeCodeCamp/News
FreeCodeCamp/NonprofitProjects
FreeCodeCamp/TeamViewer
FreeCodeCamp/Wiki
FreeCodeCamp/GameDev
FreeCodeCamp/Gaming
FreeCodeCamp/Hardware
FreeCodeCamp/Music
FreeCodeCamp/Saving
FreeCodeCamp/SelfImprovement
FreeCodeCamp/TVandMovies

Should the bot be in any of these rooms?
cc: @QuincyLarson & @BerkeleyTrue 

Let me know if this isn't right and I can tweak it more.

abhisekp commented on nap/data/RoomData.js:
```diff
@@ -25,14 +25,6 @@ var AppConfig = require('../config/AppConfig');
  * @param {string|Array<string>} roomNames List of room names or a single room name
```
These comments are documentation for the followed function. They can be removed.

joepurdy commented:
@abhisekp Doh! You're completely right, I've updated my branch with a new commit removing that documentation

SaintPeter commented:
These rooms should be removed:
```js
"dcsan/botzy",
"dcsan/gitterbot"
"camperbot/HelpZiplines",
"camperbot/devteam",
"camperbot/testing",
```
Add: 
```js
FreeCodeCamp/camperbot
FreeCodeCamp/admin
```

BerkeleyTrue commented:
@joepurdy Please remove those rooms not on the official list.

joepurdy commented:
I updated the rooms list to match the official list and added the new camperbot rooms.

Assuming no one has any objections this is ready to merge!

cc: @BerkeleyTrue

SaintPeter commented:
Looks good to me, for now.  We can always revisit it.

abhisekp commented on nap/data/RoomData.js:
```diff
@@ -128,476 +108,28 @@ var bonfireDashedNames = [
+    "FreeCodeCamp/Casual",
+    "FreeCodeCamp/CodeReview",
+    "FreeCodeCamp/CodingJobs",
+    "FreeCodeCamp/CurriculumDevelopment",
```
@BerkeleyTrue Is a bot needed in the curriculum dev room?

BerkeleyTrue replied:
Don't see any harm in it for now.
